src/items.rs

```diff
@@ -1811,12 +1825,36 @@ fn rewrite_ty<R: Rewrite>(
             _ => format!("{result}="),
         };
 
-        // 1 = `;`
-        let shape = Shape::indented(indent, context.config).sub_width(1)?;
-        rewrite_assign_rhs(context, lhs, &*ty, &RhsAssignKind::Ty, shape).map(|s| s + ";")
+        // 1 = `;` unless there's a trailing where clause
+        let shape = if after_where_predicates.is_empty() {
+            Shape::indented(indent, context.config).sub_width(1)?
```

Comment:
```suggestion
        let shape = if after_where_predicates.is_empty() {
            Shape::indented(indent, context.config).sub_width(const { ";".len() })?
```

/half serious

Reply:
would rather do that for all the magic numbers in rustfmt all at once

Reply:
the width adjustment numbers aren't always based off a static string, and i'd rather stick with direct numbers and inline comments explaining where it's coming from, if for no reason other than succinctness